src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -246,6 +248,8 @@ public void Reset()
 
             _manuallySetRequestAbortToken = null;
             _abortedCts = null;
+
+            _requestRejectReason = RequestRejectionReason.None;
```

Comment:
Why does `_requestRejectReason` need to be reset, when `_corruptedRequest` was not?